### DISCUSSION TITLE
Vue injected code blocks are not syntax highlighted

### DISCUSSION THREAD

**Discussion by Kurohyou** [vuejs/vitepress#3019] [Q&A]
I have a documentation site using some code blocks that are injected via vue (shown with quote syntax as code syntax presents showing markdown code marker):

> \#\#\# Example
> \`\`\`pug-vue
> {{obj.example}}
> \`\`\`
> 
> </div>
> <div v-if="obj.output && !/^(kscript|module)$/.test(obj.name)">
> 
> \#\#\# Output
> \`\`\`html-vue
> {{obj.output}}
> \`\`\`

However, in the site, the syntax highlighting does not trigger for this appropriately:

![image](https://github.com/vuejs/vitepress/assets/18427622/01f54e72-ccdb-4782-8db0-6c0ad82d1f29)

I have other code blocks on the site that are hardcoded, and they work correctly:

> \`\`\`pug
> +input({name:'my attribute',type:'text'})
> +button({name:'my button',type:'action'})
> \`\`\`

![image](https://github.com/vuejs/vitepress/assets/18427622/a410a3bd-c6ec-4db0-a1eb-f6dd26cb23e8)

Any idea what might be causing the failure of the syntax highlighting in the Vue injected code?

**Comment by brc-dd** [comment]
It's already documented that syntax highlighting doesn't work for those.

**Reply by Kurohyou** [reply]
Can you direct me to where that documentation is?

**Reply by brc-dd** [reply]
Exactly below where you saw `-vue` syntax: https://vitepress.dev/guide/using-vue#unescape-in-code-blocks

<img width="677" alt="image" src="https://github.com/vuejs/vitepress/assets/40380293/7e320f80-4f4d-47dd-ae0e-4c7afbe1b114">
